core/types/crosslink.go

```diff
@@ -130,17 +130,25 @@ type CrossLinks []CrossLink
 // Sort crosslinks by shardID and then tie break by blockNum then by viewID
 func (cls CrossLinks) Sort() {
 	sort.Slice(cls, func(i, j int) bool {
-		return cls[i].ShardID() < cls[j].ShardID() ||
-			(cls[i].ShardID() == cls[j].ShardID() && cls[i].Number().Cmp(cls[j].Number()) < 0) ||
-			(cls[i].ShardID() == cls[j].ShardID() && cls[i].Number() == cls[j].Number() && cls[i].ViewID().Cmp(cls[j].ViewID()) < 0)
```

Comment:
Comparing cls[i].Number() == cls[j].Number() is always false

Reply:
why it's always false?

Reply:
Right. Thanks.